package/src/middleware.ts

```diff
@@ -0,0 +1,17 @@
+import { defineMiddleware } from "astro:middleware";
+import { getEnvVariable } from "virtual:astro-env/entrypoint";
+
+export const onRequest = defineMiddleware(async ({ locals }, next) => {
+	locals.env = new Proxy<Record<string, string>>(
+		{},
+		{
+			get(_target, prop, _receiver) {
+				if (typeof prop === "string") {
+					return getEnvVariable(prop, { locals });
+				}
+			},
```

Comment:
Writing to this proxy will be silently ignored.

```
> p = new Proxy({}, {get: (_, prop) => prop})
Proxy [ {}, { get: [Function: get] } ]
> p.foo
'foo'
> p.foo = 123
123
> p.foo
'foo'
```

I think it should either be a true write, allowing envs to be modified or it should throw an error.

```suggestion
			get(target, prop, _receiver) {
				if (typeof prop === "string") {
					return target[prop] ?? getEnvVariable(prop, { locals });
				}
			},
```

Reply:
I'm not familiar with proxies at all tbh! Any improvement is more than welcome